### DISCUSSION TITLE
Re-normalization of 'RNA' assay after integration before findMarker and other visualization

### DISCUSSION THREAD

**Discussion by stevehxf** [satijalab/seurat#5209] [Q&A]
Hi Seurat team,

 According to the updated vignette of integration (https://satijalab.org/seurat/articles/integration_introduction.html), after integration, one should only change the active assay to 'RNA' before differential expression analysis and gene expression visualization.  In the old vignette, if I remember correctly, re-normalization of the RNA assay is required. Without re-normalization, my visualization is in linear scale rather than log-scale. Could you please help?

Thanks very much!

Steve

**Comment by cswoboda** [comment]
Hi Steve,

Not super clear to me what you need help with. Is there a reason you don't want to renormalize? I'm not sure how you would effectively use a log scale without normalized data.

**Comment by stevehxf** [comment]
Hi,

I meant the vignette may need to tell people to do normalization after switch the assay from 'integrated' to 'RNA'. Let me know if I am wrong.

**Reply by cswoboda** [reply]
You're correct, depending on the case. It's not specifically stated in the SCTransform integrated vignette, so if you're integrating with the SCTransform workflow they don't mention specifically to switch and renormalize so I do think that would be a useful addition. If you're using the standard workflow you don't have to renormalize, as your data was normalized prior to integration! You just need to switch assays.

Hope that helps, I agree Seurat team may want to add an explicit mention of using the RNA assay and normalizing in the SCTransform vignette in the future.

Casey

**Reply by Erinke89** [reply]
Dear @cswoboda , thank you for the clarification. Sorry if this is a silly question, but does this mean that when I want to visualize gene expression and/or do differential expression analysis after performing integration on SCTransform-processed data, that I'd better go back and make SCT my default assay instead of the RNA assay ?
Many thanks!

**Reply by saketkc** [reply]
With the v2 normalization, you can use the SCT assay for visualization and differential expression. See this [vignette](https://satijalab.org/seurat/articles/sctransform_v2_vignette.html) for a workflow.

**Reply by afuiten** [reply]
Hello, as a follow-up question, if I want to get DEGs for individual clusters (via FindAllMarkers()) after performing integration on SCTransform-processed data, should I use the SCT assay (vs the RNA assay)? I ask this because the “Introduction to SCTransform, v2 regularization” [vignette](https://satijalab.org/seurat/articles/sctransform_v2_vignette.html) addresses how to do differential expression across conditions, but I’m interested instead in finding the markers that define the clusters.

For context, I integrated two biological replicates from a single experimental timepoint, so I’m just interested in finding the markers defining the clusters and not interested in performing differential expression between the two replicates I integrated. I followed the workflow on the “Introduction to SCTransform, v2 regularization” vignette linked in the previous comment (using sctransform version 0.3.3).

**Reply by saketkc** [reply]
You should be able to use the SCT assay with FindMarkers() if they are v2 regularized (after running [PrepSCTFindMarkers](https://satijalab.org/seurat/reference/PrepSCTFindMarkers.html) .

**Reply by Erinke89** [reply]
What about doing dimensional reduction (specifically, PHATE) after integration after SCT v2? Am I correct to use SCT@scale.data as input? (as it was mentioned here https://github.com/satijalab/seurat/issues/2906 to use the scale.data slot). I'm sorry that I'm getting a bit lost here. Many thanks for the clarification!

**Reply by saketkc** [reply]
Yes, you can proceed with scaled data post SCT for dimensionality reduction.

**Reply by Erinke89** [reply]
Thank you for the quick response! Sorry to be annoying but this means I'm _not_ using integrated@scale.data ? Because I didnt only do SCT I also did integration of 4 samples.
